solr/contrib/ltr/src/java/org/apache/solr/ltr/ranking/Feature.java

```diff
@@ -141,14 +146,29 @@ public void setId(int id) {
     this.id = id;
   }
 
+  public static Feature fromMap(SolrResourceLoader solrResourceLoader,
+      Map<String,Object> featureMap) {
+    final String type =
+        (String) featureMap.get(CLASS_KEY);
+
+    final String name =
+        (String) featureMap.get(NAME_KEY);
+
+    @SuppressWarnings("unchecked")
+    final Map<String,Object> params =
+    (Map<String,Object>) featureMap.get(PARAMS_KEY);
+
+    return Feature.getInstance(solrResourceLoader,
+        type, name, params);
+  }
+
   protected abstract LinkedHashMap<String,Object> paramsToMap();
 
-  public LinkedHashMap<String,Object> toMap(String storeName) {
-    final LinkedHashMap<String,Object> o = new LinkedHashMap<>(4, 1.0f);
-    o.put((String)CommonLTRParams.FEATURE_NAME, name);
-    o.put((String)CommonLTRParams.FEATURE_CLASS, getClass().getCanonicalName());
-    o.put((String)CommonLTRParams.FEATURE_STORE, storeName);
-    o.put((String)CommonLTRParams.FEATURE_PARAMS, paramsToMap());
+  public LinkedHashMap<String,Object> toMap() {
```

Comment:
I agree that moving to/fromMap out of Feature.java makes sense if/since there is only the one calling class for these functions. The same would apply to the existing Normalizer.java class then.

So I would suggest to in this pull request here place the to/fromMap functions in Feature.java (consistent with the also pluggable Normalizer.java class having to/fromMap functions) and then for a separate change/pull request to factor the to/fromMap functions out of both Feature.java and Normalizer.java into the calling classes. How does that sound?